assert/assertions.go

```diff
@@ -212,19 +214,29 @@ the problem actually occurred in calling code.*/
 func CallerInfo() []string {
 
 	var pc uintptr
-	var ok bool
 	var file string
 	var line int
 	var name string
 
 	callers := []string{}
-	for i := 0; ; i++ {
-		pc, file, line, ok = runtime.Caller(i)
-		if !ok {
-			// The breaks below failed to terminate the loop, and we ran off the
-			// end of the call stack.
-			break
-		}
+	pcs := make([]uintptr, stackFrameBufferSize)
+	offset := 1
+	n := runtime.Callers(offset, pcs)
+	maybeMore := true
```

Comment:
nit: `maybeMore := n == stackFrameBufferSize` and you can tuck it behind the n==0 early return a couple lines down